x-pack/legacy/plugins/maps/common/descriptor_types.d.ts

```diff
@@ -119,3 +118,36 @@ export type VectorLayerDescriptor = LayerDescriptor & {
   joins?: JoinDescriptor[];
   style?: unknown;
 };
+
+export type RangeFieldMeta = {
+  min: number;
+  max: number;
+  delta: number;
+  isMinOutsideStdRange?: boolean;
+  isMaxOutsideStdRange?: boolean;
+};
+
+export type CategoryMeta = {
+  key: string;
+  count: number;
+};
+
+export type CategoryFieldMeta = {
+  categories: CategoryMeta[];
+};
+
+export type GeometryTypes = {
+  isPointsOnly: boolean;
+  isLinesOnly: boolean;
+  isPolygonsOnly: boolean;
+};
+
+export type StyleMetaValues = {
```

Comment:
`StyleMetaValues`, why is `Values` on the end? How about just `StyleMeta`

Reply:
otherwise conflict with name of StyleMeta class-name

Reply:
Then this is a good reason to call it `StyleMetaDescriptor`. When I originally make the comment on the name, there was no `StyleMeta` class, so no naming conflict. In cases where there is a naming conflict with an object instance then we should stick to the pattern of suffixing with `Descriptor`.